pkg/resources/apiserver/deployment.go

```diff
@@ -410,6 +407,32 @@ func getApiserverFlags(data *resources.TemplateData, etcdEndpoints []string, ena
 	return flags, nil
 }
 
+// getApiserverOverrideFlags creates all settings that may be overridden by cluster specific componentsOverrideSettings
+// otherwise global overrides or defaults will be set
+func getApiserverOverrideFlags(data *resources.TemplateData) (kubermaticv1.APIServerSettings, error) {
+	settings := kubermaticv1.APIServerSettings{}
+	// nodePortRange section
+	settings.NodePortRange = data.NodePortRange()
+	overrideNodePortRange := data.Cluster().Spec.ComponentsOverride.Apiserver.NodePortRange
+	if overrideNodePortRange != "" {
+		if _, err := net.ParsePortRange(overrideNodePortRange); err != nil {
+			return kubermaticv1.APIServerSettings{}, fmt.Errorf("node port range format is invalid: %v", err)
+		}
+		settings.NodePortRange = overrideNodePortRange
+	}
+	if settings.NodePortRange == "" {
```

Comment:
So if the NodePortRange has been specified by the controller, and the overrideNodePortRange was specified by the user, what will happen is, the overriding value will be ignored. IMO if the user specify the NodePortRange via the override field, then this value should have the upper hand, not the default value.

Reply:
So first the "kubermatic wide" value will be set, which can be overridden with the cluster specific one.
If both of them are empty the default range will be set.

So the order is clusterOverride > kubermaticOverride > default I think

Reply:
What I meant is, if the "kubermatic wide" node port was empty then the override node port value will be ignored. Even if the overrideNodePortRange was not empty.

Reply:
Regardless of if the "kubermatic wide" value will be set, the settings.NodePortRange will be set to `overrideNodePortRange` as long as it is not empty and a valid port range in line 421

I am not quite sure what I am overlooking here

Reply:
Ahh sorry mate, I am the one who overlooked stuff. You are right.